### PR TITLE
Add whitespace control to "Permissions required for GitHub Apps"

### PR DESCRIPTION
**Pull request by Simran-B**
### Why:

[Permissions required for GitHub Apps](https://docs.github.com/en/github-ae@latest/rest/reference/permissions-required-for-github-apps) contains several endpoint lists, but some are displayed dense and others widely spaced. This is caused by unintentional blank lines in the list markup, namely by version conditions using `{% if %}` blocks without whitespace control.

### What's being changed:

- Add whitespace control, e.g. `{% if -%}` (note the hyphen)
- Simplify logic for `Permission on "security events"` section
- Minor consistency changes (`-` for list items instead of `*`), no blank lines between items

### Check off the following:

- [x] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).

### CODE REVIEW

**Comment by janiceilene** [comment]
Thanks so much for opening a PR with this fix @Simran-B! I'll get it triaged for review ⚡

File: content/rest/reference/permissions-required-for-github-apps.md [hunk 1]
```diff
@@ -399,14 +399,19 @@ _Import_
...
 - [`DELETE /repos/:owner/:repo/comments/:comment_id/reactions/:reaction_id`](/rest/reference/reactions#delete-a-commit-comment-reaction) (:write)
 - [`DELETE /repos/:owner/:repo/issues/:issue_number/reactions/:reaction_id`](/rest/reference/reactions#delete-an-issue-reaction) (:write)
 - [`DELETE /repos/:owner/:repo/issues/comments/:comment_id/reactions/:reaction_id`](/rest/reference/reactions#delete-an-issue-comment-reaction) (:write)
 - [`DELETE /repos/:owner/:repo/pulls/comments/:comment_id/reactions/:reaction_id`](/rest/reference/reactions#delete-a-pull-request-comment-reaction) (:write)
 - [`DELETE /orgs/:org/teams/:team_slug/discussions/:discussion_number/reactions/:reaction_id`](/rest/reference/reactions#delete-team-discussion-reaction) (:write)
-- [`DELETE /orgs/:org/teams/:team_slug/discussions/:discussion_number/comments/:comment_number/reactions/:reaction_id`](/rest/reference/reactions#delete-team-discussion-comment-reaction) (:write){% endif %}
+- [`DELETE /orgs/:org/teams/:team_slug/discussions/:discussion_number/comments/:comment_number/reactions/:reaction_id`](/rest/reference/reactions#delete-team-discussion-comment-reaction) (:write)
+{% endif -%}
```
Comment (Simran-B) [inline content/rest/reference/permissions-required-for-github-apps.md:414]: Actually need whitespace here or _Releases_ ends up as part of the previous list
```suggestion
{% endif %}
```

File: content/rest/reference/permissions-required-for-github-apps.md [hunk 2]
```diff
@@ -360,9 +360,9 @@ _Branches_
 - [`POST /repos/:owner/:repo/branches/:branch/protection/restrictions/apps`](/rest/reference/repos#add-app-access-restrictions) (:write)
 - [`PUT /repos/:owner/:repo/branches/:branch/protection/restrictions/apps`](/rest/reference/repos#set-app-access-restrictions) (:write)
 - [`DELETE /repos/:owner/:repo/branches/:branch/protection/restrictions/apps`](/rest/reference/repos#remove-user-access-restrictions) (:write)
-{% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@3.0" or currentVersion == "github-ae@latest" %}
+{% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@3.0" or currentVersion == "github-ae@latest" -%}
 - [`POST /repos/:owner/:repo/branches/:branch/rename`](/rest/reference/repos#rename-a-branch) (:write)
-{% endif %}
+{% endif -%}
```
Comment (Simran-B) [inline content/rest/reference/permissions-required-for-github-apps.md:365]: Actually need whitespace here or _Commit comments_ ends up as part of the previous list
```suggestion
{% endif %}
```

File: content/rest/reference/permissions-required-for-github-apps.md [hunk 3]
```diff
@@ -212,9 +212,9 @@ _Branches_
 - [`POST /repos/:owner/:repo/branches/:branch/protection/restrictions/users`](/rest/reference/repos#add-user-access-restrictions) (:write)
 - [`PUT /repos/:owner/:repo/branches/:branch/protection/restrictions/users`](/rest/reference/repos#set-user-access-restrictions) (:write)
 - [`DELETE /repos/:owner/:repo/branches/:branch/protection/restrictions/users`](/rest/reference/repos#remove-user-access-restrictions) (:write)
-{% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@3.0" or currentVersion == "github-ae@latest" %}
+{% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@3.0" or currentVersion == "github-ae@latest" -%}
 - [`POST /repos/:owner/:repo/branches/:branch/rename`](/rest/reference/repos#rename-a-branch) (:write)
-{% endif %}
+{% endif -%}
```
Comment (Simran-B) [inline content/rest/reference/permissions-required-for-github-apps.md:217]: Actually need whitespace here or Commit comments ends up as part of the previous list

```suggestion
{% endif %}
```

File: content/rest/reference/permissions-required-for-github-apps.md [hunk 4]
```diff
@@ -767,14 +776,17 @@ _Reactions_
...
 - [`DELETE /repos/:owner/:repo/issues/comments/:comment_id/reactions/:reaction_id`](/rest/reference/reactions#delete-an-issue-comment-reaction) (:write)
 - [`DELETE /repos/:owner/:repo/pulls/comments/:comment_id/reactions/:reaction_id`](/rest/reference/reactions#delete-a-pull-request-comment-reaction) (:write)
 - [`DELETE /orgs/:org/teams/:team_slug/discussions/:discussion_number/reactions/:reaction_id`](/rest/reference/reactions#delete-team-discussion-reaction) (:write)
-- [`DELETE /orgs/:org/teams/:team_slug/discussions/:discussion_number/comments/:comment_number/reactions/:reaction_id`](/rest/reference/reactions#delete-team-discussion-comment-reaction) (:write){% else %}- [`DELETE /reactions/:reaction_id`](/rest/reference/reactions#delete-a-reaction) (:write){% endif %}
+- [`DELETE /orgs/:org/teams/:team_slug/discussions/:discussion_number/comments/:comment_number/reactions/:reaction_id`](/rest/reference/reactions#delete-team-discussion-comment-reaction) (:write)
+{% else -%}
+- [`DELETE /reactions/:reaction_id`](/rest/reference/reactions#delete-a-reaction) (:write)
+{% endif -%}
```
Comment (Simran-B) [inline content/rest/reference/permissions-required-for-github-apps.md:789]: Actually need whitespace here or _Requested reviewers_ ends up as part of above list
```suggestion
{% endif %}
```

**Comment by Simran-B** [comment]
I'll do another pass to revert some whitespace control again where it breaks the layout or is unnecessary (at the end of lists).

File: content/rest/reference/permissions-required-for-github-apps.md [hunk 5]
```diff
@@ -523,15 +530,17 @@ _Reactions_
...
 - [`DELETE /repos/:owner/:repo/issues/comments/:comment_id/reactions/:reaction_id`](/rest/reference/reactions#delete-an-issue-comment-reaction) (:write)
 - [`DELETE /repos/:owner/:repo/pulls/comments/:comment_id/reactions/:reaction_id`](/rest/reference/reactions#delete-a-pull-request-comment-reaction) (:write)
 - [`DELETE /orgs/:org/teams/:team_slug/discussions/:discussion_number/reactions/:reaction_id`](/rest/reference/reactions#delete-team-discussion-reaction) (:write)
-- [`DELETE /orgs/:org/teams/:team_slug/discussions/:discussion_number/comments/:comment_number/reactions/:reaction_id`](/rest/reference/reactions#delete-team-discussion-comment-reaction) (:write){% else %}
-- [`DELETE /reactions/:reaction_id`](/rest/reference/reactions#delete-a-reaction) (:write){% endif %}
+- [`DELETE /orgs/:org/teams/:team_slug/discussions/:discussion_number/comments/:comment_number/reactions/:reaction_id`](/rest/reference/reactions#delete-team-discussion-comment-reaction) (:write)
+{% else -%}
+- [`DELETE /reactions/:reaction_id`](/rest/reference/reactions#delete-a-reaction) (:write)
+{% endif -%}
```
Comment (Simran-B) [inline content/rest/reference/permissions-required-for-github-apps.md:543]:
```suggestion
{% endif %}
```

File: content/rest/reference/permissions-required-for-github-apps.md [hunk 6]
```diff
@@ -543,37 +552,37 @@ _Keys_
...
+{% if currentVersion == "free-pro-team@latest" -%}
 - [`PATCH /scim/v2/orgs/:org/Users/:external_identity_guid`](/rest/reference/scim#update-an-attribute-for-a-scim-user) (:write)
-{% endif %}
-{% if currentVersion == "free-pro-team@latest" %}
+{% endif -%}
+{% if currentVersion == "free-pro-team@latest" -%}
 - [`DELETE /scim/v2/orgs/:org/Users/:external_identity_guid`](/rest/reference/scim#delete-a-scim-user-from-an-organization) (:write)
-{% endif %}
+{% endif -%}
```
Comment (Simran-B) [inline content/rest/reference/permissions-required-for-github-apps.md:585]:
```suggestion
{% endif %}
```

File: content/rest/reference/permissions-required-for-github-apps.md [hunk 7]
```diff
@@ -624,15 +633,15 @@ _Teams_
...
+{% if currentVersion == "free-pro-team@latest" -%}
 - [`PUT /orgs/:org/interaction-limits`](/rest/reference/interactions#set-interaction-restrictions-for-an-organization) (:write)
-{% endif %}
-{% if currentVersion == "free-pro-team@latest" %}
+{% endif -%}
+{% if currentVersion == "free-pro-team@latest" -%}
 - [`DELETE /orgs/:org/interaction-limits`](/rest/reference/interactions#remove-interaction-restrictions-for-an-organization) (:write)
-{% endif %}
+{% endif -%}
```
Comment (Simran-B) [inline content/rest/reference/permissions-required-for-github-apps.md:644]:
```suggestion
{% endif %}
```

File: content/rest/reference/permissions-required-for-github-apps.md [hunk 8]
```diff
@@ -698,9 +707,9 @@ _Teams_
 - [`POST /repos/:owner/:repo/pages/builds`](/rest/reference/repos#request-a-github-pages-build) (:write)
 - [`GET /repos/:owner/:repo/pages/builds/:build_id`](/rest/reference/repos#get-github-pages-build) (:read)
 - [`GET /repos/:owner/:repo/pages/builds/latest`](/rest/reference/repos#get-latest-pages-build) (:read)
-{% if currentVersion == "free-pro-team@latest" %}
+{% if currentVersion == "free-pro-team@latest" -%}
 - [`GET /repos/:owner/:repo/pages/health`](/rest/reference/repos#get-a-dns-health-check-for-github-pages) (:write)
-{% endif %}
+{% endif -%}
```
Comment (Simran-B) [inline content/rest/reference/permissions-required-for-github-apps.md:712]:
```suggestion
{% endif %}
```

**Review by guntrip — APPROVED** [review]
Hi @Simran-B! Thanks so much for identifying the problem and providing a fix. This is looking much better and will be much more pleasant to edit in future! ✨ 

I'll approve and merge this now. Thanks again for your contribution!

**Pull request merged**